### PR TITLE
add a note about debugtoolbar 

### PR DESCRIPTION
**Pull request by myimages**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
 
-.. image:: https://readthedocs.org/projects/pyramid/badge/?version=master
-        :target: https://docs.pylonsproject.org/projects/pyramid/en/master
-        :alt: master Documentation Status
+.. image:: https://readthedocs.org/projects/pyramid/badge/?version=2.0-branch
+        :target: https://docs.pylonsproject.org/projects/pyramid/en/2.0-branch
+        :alt: 2.0-branch Documentation Status
 
 .. image:: https://img.shields.io/badge/irc-freenode-blue.svg
         :target: https://webchat.freenode.net/?channels=pyramid
```
Comment (stevepiercy) [inline README.rst:13]: Please revert all changes of `master` to `2.0-branch`. This PR is being made against the `master` branch.

File: docs/conf.py [hunk 2]
```diff
@@ -153,7 +154,7 @@ def nothing(*arg):
     github_url='https://github.com/Pylons/pyramid',
     # On master branch and new branch still in
     # pre-release status: true; else: false.
-    in_progress='true',
+    in_progress='false',
```
Comment (stevepiercy) [inline docs/conf.py:157]: Please revert this change. The `master` branch is "in progress".

File: docs/conf.py [hunk 3]
```diff
@@ -142,6 +142,7 @@ def nothing(*arg):
 #     'whatsnew-1.8': 'index',
 #     'whatsnew-1.9': 'index',
 #     'whatsnew-1.10': 'index',
+#     'whatsnew-2.0': 'index',
```
Comment (stevepiercy) [inline docs/conf.py:145]: This change is OK. It appears in the `2.0-branch`, so it does not need to be backported.

File: docs/narr/paste.rst [hunk 4]
```diff
@@ -68,8 +68,8 @@ path, which refers to a callable in our ``myproject`` package's ``__init__.py``
 module.
 
 The ``egg:`` prefix in ``egg:myproject`` indicates that this is an entry point
-*URI* specifier, where the "scheme" is "egg".  An "egg" is created when you run
-``setup.py install`` or ``setup.py develop`` within your project.
+*URI* specifier, where the "scheme" is "egg".
+An "egg" is created when you install your project.
```
Comment (stevepiercy) [inline docs/narr/paste.rst:72]: This change is OK.

File: docs/narr/project.rst [hunk 5]
```diff
@@ -83,7 +83,7 @@ On all platforms, generate a project using cookiecutter.
 
 .. code-block:: bash
 
-    cookiecutter gh:Pylons/pyramid-cookiecutter-starter --checkout master
+    cookiecutter gh:Pylons/pyramid-cookiecutter-starter --checkout 2.0-branch
```
Comment (stevepiercy) [inline docs/narr/project.rst:86]: Please revert.

File: docs/quick_tour.rst [hunk 6]
```diff
@@ -505,7 +505,7 @@ Let's use the cookiecutter ``pyramid-cookiecutter-starter`` to create a starter
 
 .. code-block:: bash
 
-    $VENV/bin/cookiecutter gh:Pylons/pyramid-cookiecutter-starter --checkout master
+    $VENV/bin/cookiecutter gh:Pylons/pyramid-cookiecutter-starter --checkout 2.0-branch
```
Comment (stevepiercy) [inline docs/quick_tour.rst:508]: Please revert.

File: docs/quick_tour.rst [hunk 7]
```diff
@@ -865,7 +865,7 @@ Pyramid and SQLAlchemy are great friends. That friendship includes a cookiecutte
 .. code-block:: bash
 
     cd ~
-    env/bin/cookiecutter gh:Pylons/pyramid-cookiecutter-starter --checkout master
+    env/bin/cookiecutter gh:Pylons/pyramid-cookiecutter-starter --checkout 2.0-branch
```
Comment (stevepiercy) [inline docs/quick_tour.rst:868]: Please revert.

File: docs/quick_tutorial/cookiecutters.rst [hunk 8]
```diff
@@ -28,7 +28,7 @@ Steps
 
     .. code-block:: bash
 
-        $VENV/bin/cookiecutter gh:Pylons/pyramid-cookiecutter-starter --checkout master
+        $VENV/bin/cookiecutter gh:Pylons/pyramid-cookiecutter-starter --checkout 2.0-branch
```
Comment (stevepiercy) [inline docs/quick_tutorial/cookiecutters.rst:31]: Please revert.

File: docs/quick_tutorial/debugtoolbar.rst [hunk 9]
```diff
@@ -65,7 +65,8 @@ Steps
         $VENV/bin/pserve development.ini --reload
 
 #.  Open http://localhost:6543/ in your browser.
-    See the handy toolbar on the right.
+    See the handy toolbar on the right. ``note: the Response object should contain <body> html or it won't be shown unless an error is introduced.``
```
Comment (stevepiercy) [inline docs/quick_tutorial/debugtoolbar.rst:68]: If you follow the tutorial correctly, does this issue actually occur? If not, then please remove this note.

Also as an FYI, the correct syntax for notes in reStructuredText is the following, with proper indentation:

```
.. note::

    This is my note.
    It is multi-line, one sentence per line.
```

or

```
.. note::

    This is my single sentence note on one line.
```

or

```
.. note:: This is my single sentence note on one line.
```

I prefer the first two formats to the last for all admonitions because it makes it easier to add more text, if needed.

File: docs/tutorials/modwsgi/index.rst [hunk 10]
```diff
@@ -39,7 +39,7 @@ specific path information for commands and files.
     .. code-block:: bash
 
         cd ~
-        cookiecutter gh:Pylons/pyramid-cookiecutter-starter --checkout master
+        cookiecutter gh:Pylons/pyramid-cookiecutter-starter --checkout 2.0-branch
```
Comment (stevepiercy) [inline docs/tutorials/modwsgi/index.rst:42]: Please revert.

File: docs/tutorials/wiki/installation.rst [hunk 11]
```diff
@@ -36,7 +36,7 @@ On Unix
 .. code-block:: bash
 
     cd ~
-    cookiecutter gh:Pylons/pyramid-cookiecutter-starter --checkout master
+    cookiecutter gh:Pylons/pyramid-cookiecutter-starter --checkout 2.0-branch
```
Comment (stevepiercy) [inline docs/tutorials/wiki/installation.rst:39]: Please revert.

File: docs/tutorials/wiki/installation.rst [hunk 12]
```diff
@@ -45,7 +45,7 @@ On Windows
 .. code-block:: doscon
 
     cd \
-    cookiecutter gh:Pylons/pyramid-cookiecutter-starter --checkout master
+    cookiecutter gh:Pylons/pyramid-cookiecutter-starter --checkout 2.0-branch
```
Comment (stevepiercy) [inline docs/tutorials/wiki/installation.rst:48]: Please revert.

File: docs/tutorials/wiki2/installation.rst [hunk 13]
```diff
@@ -43,15 +43,15 @@ On Unix
 .. code-block:: bash
 
     cd ~
-    cookiecutter gh:Pylons/pyramid-cookiecutter-starter --checkout master
+    cookiecutter gh:Pylons/pyramid-cookiecutter-starter --checkout 2.0-branch
```
Comment (stevepiercy) [inline docs/tutorials/wiki2/installation.rst:46]: Please revert.

File: docs/tutorials/wiki2/installation.rst [hunk 14]
```diff
@@ -43,15 +43,15 @@ On Unix
...
 
 On Windows
 ^^^^^^^^^^
 
 .. code-block:: doscon
 
     cd \
-    cookiecutter gh:Pylons/pyramid-cookiecutter-starter --checkout master
+    cookiecutter gh:Pylons/pyramid-cookiecutter-starter --checkout 2.0-branch
```
Comment (stevepiercy) [inline docs/tutorials/wiki2/installation.rst:54]: Please revert.

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Please see my individual comments and make changes. Thank you!

**Comment by mmerickel** [comment]
@stevepiercy could you please delete your comments on each line in this PR. They don't need to revert the changes, they need to rebase their commit onto the correct branch and push again. If you look at all of the commits pushed in this PR it's clear the change was made on 2.0-branch and then submitted to master, bringing in all of the changes from the 2.0-branch along with it.

**Comment by stevepiercy** [comment]
@mmerickel ah, OK. Done!

File: docs/quick_tutorial/debugtoolbar.rst (hunk 9 above)
Comment (myimages) [reply docs/quick_tutorial/debugtoolbar.rst:68]: tutorial approach suggests **copying** the previous folder of the tutorial to the next steps of the tutorial. 

https://docs.pylonsproject.org/projects/pyramid/en/latest/quick_tutorial/tutorial_approach.html

so I made that change in previous steps of the tutorial ( which was a few days ago ). 

when I refreshed the page I didn't see the debugtoolbar and I already **fogot**  that I made this small change in the response string.( I am just discovering the framework so I don't want any html in my response. ) 

but you know beginners want to see something that works and build upon and that might prevent new comers from trying different things to get it to work given that they are new.

a suggested edit also will be introducing an error with `Xresponse` instead of `Response` as a step of the tutorial not as an extra credit.

Comment (stevepiercy) [reply docs/quick_tutorial/debugtoolbar.rst]: > tutorial approach suggests copying the previous folder of the tutorial to the next steps of the tutorial.

In the Quick Tutorial, it's a _requirement_ to copy the previous step for all but a few steps. You'll see why when the tutorial forks from a much earlier step.

> when I refreshed the page I didn't see the debugtoolbar and I already fogot that I made this small change in the response string.( I am just discovering the framework so I don't want any html in my response. )

If you experiment or make mistakes, then that's great for your learning. However we will not include experiments in the tutorial documentation. It is intended to be a "_Quick_ Tutorial". We also include the source code in the Pyramid repository, and we recommend that users clone it so that they can compare their code with official source code to find their mistakes.

FYI, the debugtoolbar is intended to be used only in development environments. The [Usage documentation](https://docs.pylonsproject.org/projects/pyramid_debugtoolbar/en/latest/#usage) states:

> For any HTML responses generated by the application, a link to the toolbar for the current page will be available in the upper right corner, provided the response contains a closing `</body>` tag.

We prefer to reference documentation with links to it instead of duplicating it.

> but you know beginners want to see something that works and build upon and that might prevent new comers from trying different things to get it to work given that they are new.

There is a future step, [14: AJAX Development With JSON Renderers](https://docs.pylonsproject.org/projects/pyramid/en/latest/quick_tutorial/json.html) that covers an example of how to return a JSON response with a content-type of `application/json` instead of the default content-type of `text/html`.

We have a [few more official tutorials](https://docs.pylonsproject.org/projects/pyramid/en/latest/index.html#tutorials), and there are some [community maintained tutorials](https://docs.pylonsproject.org/projects/pyramid_tutorials/en/latest/). There is also a very active project [pyramid_openapi3](https://github.com/pylons/pyramid_openapi3) that provides [several demo apps](https://github.com/pylons/pyramid_openapi3#demo--examples).

> a suggested edit also will be introducing an error with Xresponse instead of Response as a step of the tutorial not as an extra credit.

I'm not sure how that would fit into the Quick Tutorial. I'd have to see a draft of the proposal. It might be a better candidate for the [Pyramid Community Cookbook](https://docs.pylonsproject.org/projects/pyramid_cookbook/en/latest/).

There's also a great [community](https://trypyramid.com/community.html) as a resource for new developers. We were all new developers to Pyramid at one point, so we know how it is.

File: docs/quick_tutorial/debugtoolbar.rst [hunk 15]
```diff
@@ -65,7 +65,11 @@ Steps
         $VENV/bin/pserve development.ini --reload
 
 #.  Open http://localhost:6543/ in your browser.
-    See the handy toolbar on the right. ``note: the Response object should contain <body> html or it won't be shown unless an error is introduced.``
+    See the handy toolbar on the right. 
+    
+      
+      .. note:: 
+        the Response object should contain ``<body></body>`` tag , so if you have changed this in previous steps of the tutorial make sure that you include the it         in order for this to work.
```
Comment (stevepiercy) [inline docs/quick_tutorial/debugtoolbar.rst:72]: This is a duplicate of debugtoolbar documentation, and we already link to it in the Quick Tutorial under [Usage](https://docs.pylonsproject.org/projects/pyramid_debugtoolbar/en/latest/#usage). I'm sorry, but I cannot approve this change.

**Review by stevepiercy — CHANGES_REQUESTED** [review]

File: docs/quick_tutorial/debugtoolbar.rst (hunk 15 above)
Comment (myimages) [reply docs/quick_tutorial/debugtoolbar.rst:72]: this is a **quick tutorial** not a documentation , the documentation is comprehensive but who is doing a **quick tutorial**  then goes to a specific documentation about debugtoolbar when you have nothing to debug and hasn't even started using the framework ? you know where I am going with this ?

Comment (myimages) [reply docs/quick_tutorial/debugtoolbar.rst:72]: this specific change doesn't have to be merged , the point is to make the documentation more easy to follow and account for such a change. 

that is the point of opening this PR.

**Comment by bertjwregeer** [comment]
I've updated the base branch from `master` to `2.0-branch`.

**Comment by mmerickel** [comment]
something isn't right about this PR. Not sure if it's because of changing to 2.0-branch by @bertjwregeer or something with the original PR but it's not mergeable in its current state.

**Comment by bertjwregeer** [comment]
It looks like someone merged master into it...

**Comment by mmerickel** [comment]
Apologies @myimages but I'm going to close this as stale. The current PR has extra commits in it that are clearly not intended to be part of the PR.
